File: backend/src/Microservices/PackageSending/PackageSending.DAL/Entities/ShippingRequest.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PackageSending.DAL.Entities
{
    public class ShippingRequest
    {
        public string Id { get; set; }
        public string UserId { get; set; }
        public string CourierId { get; set; }
        public Address AddressFrom { get; set; }
        public Address AddressTo { get; set; }
        public bool IsExpress { get; set; }
        public int ShippingOptionId { get; set; }
        public int PaymentOptionId { get; set; }
        public string BillingId { get; set; }
        public bool IsFinished { get; set; }

        public PaymentOption PaymentOption { get; set; }
        public ShippingOption ShippingOption { get; set; }
        public Billing Billing { get; set; }
    }
}

File: frontend/mobil/parceldelivery_mobile/lib/models/shipping_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:parceldelivery_mobile/models/address.dart';
import 'package:parceldelivery_mobile/models/billing.dart';
import 'package:parceldelivery_mobile/models/package.dart';
import 'package:parceldelivery_mobile/models/payment_option.dart';
import 'package:parceldelivery_mobile/models/shipping_option.dart';

part 'shipping_request.freezed.dart';
part 'shipping_request.g.dart';

@freezed
class ShippingRequest with _$ShippingRequest {
  const factory ShippingRequest({
    required String id,
    required String userId,
    required String name,
    required String email,
    required String courierId,
    required Address addressFrom,
    required Address addressTo,
    required bool isExpress,
    required bool isFinished,
    required int status,
    required PaymentOption paymentOption,
    required ShippingOption shippingOption,
    required Billing billing,
    required List<Package> packages,
  }) = _ShippingRequest;

  factory ShippingRequest.fromJson(Map<String, Object?> json) =>
      _$ShippingRequestFromJson(json);
}
